src/engine/strat_engine/names.rs

```diff
@@ -18,6 +18,53 @@ use crate::{
 
 const FORMAT_VERSION: u16 = 1;
 
+/// Prefix for the key descriptions added into the kernel keyring to indicate
+/// that they were added by Stratis.
+fn key_description_prefix() -> String {
+    format!("stratis-{}-key-", FORMAT_VERSION)
+}
+
+/// A data type respresenting a key description of a key added by stratisd.
+#[derive(Debug, PartialEq)]
+pub struct KeyDescription(String);
+
+impl KeyDescription {
+    /// Check if the system key description has the Stratis prefix. If so,
+    /// return `Some` with the prefix stripped. If not, return `None`.
+    pub fn from_system_key_desc(raw_key_desc: &str) -> Option<KeyDescription> {
+        let mut key_desc = raw_key_desc.to_string();
+        let prefix = key_description_prefix();
+        if key_desc.starts_with(prefix.as_str()) {
+            key_desc.replace_range(..prefix.len(), "");
+            if key_desc.is_empty() {
+                None
+            } else {
+                Some(KeyDescription(key_desc))
+            }
+        } else {
+            None
+        }
+    }
+
+    /// Return the application-level key description (the key description with no
+    /// Stratis prefix added).
+    pub fn as_application_str(&self) -> &str {
+        &self.0
+    }
+}
+
+impl Display for KeyDescription {
```

Comment:
You might want not to implement Display since the format is not truley canonica. Maybe a ```to_system_key_desc``` method would be a better choice.